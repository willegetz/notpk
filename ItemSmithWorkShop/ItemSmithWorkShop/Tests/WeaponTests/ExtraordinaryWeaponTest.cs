using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using AdventureItems;
using ApprovalTests;
using ItemSmithWorkShop.AdventureItems.WeaponAdons;
using ItemSmithWorkShop.AdventureItems;
using ItemSmithWorkShop.AdventureItems.ExtraordinaryWeaponUtilities;

namespace ItemSmithWorkShop.Tests.WeaponTests
{
	[TestClass]
	public class ExtraordinaryWeaponTest
	{
		[TestMethod]
		public void TestCreateMasterWorkWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon masterworkWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Masterwork");
			Approvals.Approve(masterworkWeapon.GetItem());
		}

		[TestMethod]
		public void TestCreateAdamantineWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon adamantineWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Adamantine");
			Approvals.Approve(adamantineWeapon.GetItem());
		}

		[TestMethod]
		public void TestCreateAlchemicalSilverWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon silverWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Silver");
			Approvals.Approve(silverWeapon.GetItem());
		}

		[TestMethod]
		public void TestCreateColdIronWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon ironWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Cold Iron");
			Approvals.Approve(ironWeapon.GetItem());
		}

		[TestMethod]
		public void TestCreateDarkwoodWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon darkwoodWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Darkwood");
			Approvals.Approve(darkwoodWeapon.GetItem());
		}

		[TestMethod]
		public void TestCreateMithralWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon mithralWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Mithral");
			Approvals.Approve(mithralWeapon.GetItem());
		}
	}
}
