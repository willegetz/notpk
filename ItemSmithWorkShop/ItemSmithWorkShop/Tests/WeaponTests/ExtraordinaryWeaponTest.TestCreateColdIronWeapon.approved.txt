Cold Iron Dagger:	'4 gp'
Weight: '1 pound(s)'
To Hit: ''
Damage: '1d4'
Hardness: '10'
Hit Points: '2'
	Cold Iron is effective against some Fey.
	Magic enhancements cost an additional 2000 gold pieces.
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.